packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -22,89 +23,183 @@ import { unlock } from '../lock-unlock';
+ * @param {string} props.attrName   - The attribute name.
+ * @param {any}    props.attrValue  - The attribute value.
+ * @param {Object} props.blockProps - The block props with bound attribute.
+ * @param {Object} props.source     - Source handler.
+ * @param {Object} props.args       - The arguments to pass to the source.
+ * @return {null}                     This is a data-handling component. Render nothing.
+ */
+const BlockBindingConnector = ( {
+	args,
+	attrName,
+	attrValue,
+	blockProps,
+	source,
+} ) => {
+	const { placeholder, value: propValue } = source.useSource(
+		blockProps,
+		args
+	);
+
+	const blockName = blockProps.name;
+
+	const setAttributes = blockProps.setAttributes;
+	const updateBoundAttibute = useCallback(
+		( newAttrValue ) =>
+			setAttributes( {
+				[ attrName ]: newAttrValue,
+			} ),
+		[ attrName, setAttributes ]
+	);
+
+	useEffect( () => {
+		if ( typeof propValue !== 'undefined' ) {
+			updateBoundAttibute( propValue );
+		} else if ( placeholder ) {
+			/*
+			 * Placeholder fallback.
+			 * If the attribute is `src` or `href`,
+			 * a placeholder can't be used because it is not a valid url.
+			 * Adding this workaround until
+			 * attributes and metadata fields types are improved and include `url`.
+			 */
+			const htmlAttribute =
+				getBlockType( blockName ).attributes[ attrName ].attribute;
+
+			if ( htmlAttribute === 'src' || htmlAttribute === 'href' ) {
+				updateBoundAttibute( null );
+				return;
+			}
+
+			updateBoundAttibute( placeholder );
+		}
+	}, [
+		updateBoundAttibute,
+		propValue,
+		attrValue,
+		placeholder,
+		blockName,
+		attrName,
+	] );
+
+	return null;
+};
+
+function BlockBindingBridge( { bindings, props } ) {
+	if ( ! bindings ) {
+		return null;
+	}
+
+	const { name, attributes } = props;
+
+	// Collect all the binding connectors.
+	const BindingConnectorInstances = [];
+
+	Object.entries( bindings ).forEach( ( [ attrName, boundAttribute ], i ) => {
+		// Check if the block attribute can be bound.
+		if ( ! hasPossibleBlockBinding( name, attrName ) ) {
+			return;
+		}
+
+		const { getBlockBindingsSource } = unlock( select( blockEditorStore ) );
+
+		// Bail early if the block doesn't have a valid source handler.
+		const source = getBlockBindingsSource( boundAttribute.source );
+		if ( ! source ) {
+			return;
+		}
+
+		BindingConnectorInstances.push(
+			<BlockBindingConnector
+				key={ `${ boundAttribute.source }-${ name }-${ attrName }-${ i }` }
+				attrName={ attrName }
+				attrValue={ attributes[ attrName ] }
+				source={ source }
+				blockProps={ props }
+				args={ boundAttribute.args }
+			/>
+		);
+	} );
+
+	return <>{ BindingConnectorInstances }</>;
+}
+
+const withBlockBindingSupport = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
+		const { attributes } = props;
+
+		// Bail early if the block doesn't have bindings.
+		const bindings = attributes?.metadata?.bindings;
+		if ( ! bindings ) {
+			return null;
```

Comment:
This filter should always return the `BlockEdit`; otherwise, blocks won't render.

Reply:
updated (what a shame) sorry about it.